app/services/group_finalizer.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+#
+# Service object to finalize groups
+class GroupFinalizer < Updater
+  def self.finalize(**params)
+    new(**params).finalize
+  end
+
+  # Initialize a GroupFinalizer
+  #
+  # @param group [Group] The group object to be updated
+  def initialize(group:)
+    @group = group
+    @errors = []
+  end
+
+  def finalize
+    return error unless valid?
+    ActiveRecord::Base.transaction do
+      group.finalizing!
+      group.leader.membership.update(locked: true)
+    end
+    success
+  rescue ActiveRecord::RecordInvalid => failures
+    @errors = failures
+    error
+  end
+
+  private
+
+  attr_reader :group, :errors
+
+  def valid?
+    group_full
+    suite_size_available
+    return false unless errors.empty?
+    true
+  end
+
+  def group_full
```

Comment:
Shouldn't this be `group_full?`? I know it's a minor issue but these feel like query methods to me.